app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -499,6 +504,175 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
+                defaultMozBrowser.set(it)
+            }
+
+            mozillaProducts.set(mozillaProductDetector.getInstalledMozillaProducts(applicationContext))
+
+            adjustCampaign.set(settings.adjustCampaignId)
+            adjustAdGroup.set(settings.adjustAdGroup)
+            adjustCreative.set(settings.adjustCreative)
+            adjustNetwork.set(settings.adjustNetwork)
+
+            searchWidgetInstalled.set(settings.searchWidgetInstalled)
+
+            val openTabsCount = settings.openTabsCount
+            hasOpenTabs.set(openTabsCount > 0)
+            if (openTabsCount > 0) {
+                tabsOpenCount.add(openTabsCount)
+            }
+
+            val topSitesSize = settings.topSitesSize
+            hasTopSites.set(topSitesSize > 0)
+            if (topSitesSize > 0) {
+                topSitesCount.add(topSitesSize)
+            }
+
+            val installedAddonSize = settings.installedAddonsCount
+            Addons.hasInstalledAddons.set(installedAddonSize > 0)
+            if (installedAddonSize > 0) {
+                Addons.installedAddons.set(settings.installedAddonsList.split(','))
+            }
+
+            val enabledAddonSize = settings.enabledAddonsCount
+            Addons.hasEnabledAddons.set(enabledAddonSize > 0)
+            if (enabledAddonSize > 0) {
+                Addons.enabledAddons.set(settings.enabledAddonsList.split(','))
+            }
+
+            val desktopBookmarksSize = settings.desktopBookmarksSize
+            hasDesktopBookmarks.set(desktopBookmarksSize > 0)
+            if (desktopBookmarksSize > 0) {
+                desktopBookmarksCount.add(desktopBookmarksSize)
+            }
+
+            val mobileBookmarksSize = settings.mobileBookmarksSize
+            hasMobileBookmarks.set(mobileBookmarksSize > 0)
+            if (mobileBookmarksSize > 0) {
+                mobileBookmarksCount.add(mobileBookmarksSize)
+            }
+
+            toolbarPosition.set(
+                when (settings.toolbarPosition) {
+                    ToolbarPosition.BOTTOM -> Event.ToolbarPositionChanged.Position.BOTTOM.name
+                    ToolbarPosition.TOP -> Event.ToolbarPositionChanged.Position.TOP.name
+                }
+            )
+
+            tabViewSetting.set(settings.getTabViewPingString())
+            closeTabSetting.set(settings.getTabTimeoutPingString())
+        }
+
+        browserStore.waitForSelectedOrDefaultSearchEngine { searchEngine ->
+            if (searchEngine != null) {
+                SearchDefaultEngine.apply {
+                    code.set(searchEngine.id)
+                    name.set(searchEngine.name)
+                    submissionUrl.set(searchEngine.buildSearchUrl(""))
+                }
+            }
+        }
+    }
+
+    @Suppress("ComplexMethod")
+    private fun setPreferenceMetrics(
+        settings: Settings
+    ) {
+        // We purposefully make all of our preferences the string_list format to make data analysis
```

Comment:
I imagine we should change this comment if not remove it entirely since they're boolean now. I think this comment shouldn't be applied to the `with` block either since it only applies to line 613-623.

Reply:
Good catch.   Thanks I'll update